kolibri/utils/sanity_checks.py

```diff
@@ -0,0 +1,44 @@
+import logging
+import socket
+import sys
+
+from .server import get_status
+from .server import NotRunning
+
+logger = logging.getLogger(__name__)
+
+def check_other_kolibri_running(port):
+    try:
+        # Check if there are other kolibri instances running
+        # If there are, then we need to stop users from starting kolibri again.
+        pid, listen_address, listen_port = get_status()
+        logger.error(
+            "There is another Kolibri server running. "
+            "Please use `kolibri stop` and try again."
+        )
+        sys.exit(1)
```

Comment:
Just flagging: This is where we need to start defining the actual return codes such that UIs can easily find out why Kolibri doesn't start.

Reply:
Thanks for the information! Maybe raising the exception instead of `sys.exit(1) would be good?

Reply:
As you show in this great PR, the startup checks are specific to exactly what happens. Therefor, I would think that raising exceptions and catching them in a systematic way to make program execution terminate will be hard. The user guiding error message is very important :)

Reply:
What I mean is that here there is a port availability check, which is important to `start`. Maybe also `restart`. But it's not necessary for `manage`.

Reply:
Thank you! I just realized what you mean about the return codes. As for the `restart`, I'm sorry that I didn't notice this before. But it seems that `restart` is not really restarting the server right now?